src/stores/ui.ts

```diff
@@ -1,33 +1,148 @@
 import { defineStore } from 'pinia'
 
-export type SnackbarState = 'not_shown' | 'visible' | 'dismissed'
+/**
+ * By default, the filters are shown on desktop, and hidden on the mobile widths.
+ * `dismissed` is used when the filters are closed on the desktop width.
+ */
+export const VISIBLE = 'visible'
+export const DISMISSED = 'dismissed'
+export const NOT_SHOWN = 'not_shown'
+const snackbarStates = [NOT_SHOWN, VISIBLE, DISMISSED] as const
+export type SnackbarState = typeof snackbarStates[number]
 
+interface UiStateCookie {
+  isDesktopLayout?: boolean | undefined
+  isMobileUa?: boolean | undefined
+  isFilterDismissed?: boolean | undefined
```

Comment:
The `undefined` is redundant (sort of, but not entirely) with the `?` notation on the property name.

Are there cases where the property will exist but be set to the value `undefined`? If that is the case then this is the correct type. Otherwise, if any time the property exists it will be a boolean value, then we can leave undefined off the righthand side.

```suggestion
  isDesktopLayout?: boolean
  isMobileUa?: boolean
  isFilterDismissed?: boolean
```

Reply:
Thank you for the explanation. I've reviewed the usage and removed the unnecessary `| undefined` from here.